embedded/src/main/java/org/wildfly/core/embedded/EmbeddedServerFactory.java

```diff
@@ -338,4 +340,44 @@ private static void setupLoggingSystem(ModuleLoader moduleLoader) {
             WildFlySecurityManager.setCurrentContextClassLoaderPrivileged(tccl);
         }
     }
+
+    private static Object createServer(final Method createServerMethod, final ModuleLoader moduleLoader, final File jbossHomeDir, final String[] cmdargs) {
+        Object serverImpl;
+        try {
+            Properties sysprops = WildFlySecurityManager.getSystemPropertiesPrivileged();
+            Map<String, String> sysenv = WildFlySecurityManager.getSystemEnvironmentPrivileged();
+            String[] args = cmdargs != null ? cmdargs : new String[0];
+            serverImpl = createServerMethod.invoke(null, jbossHomeDir, moduleLoader, sysprops, sysenv, args);
+        } catch (final InvocationTargetException ite) {
+            throw EmbeddedLogger.ROOT_LOGGER.cannotCreateStandaloneServer(ite.getCause(), createServerMethod);
+        } catch (final IllegalAccessException iae) {
+            throw EmbeddedLogger.ROOT_LOGGER.cannotCreateStandaloneServer(iae, createServerMethod);
```

Comment:
This is a pre-existing issue, but might as well correct it now. These error messages are not correct in the HC case.